### DISCUSSION TITLE
Repository structure

### DISCUSSION THREAD

**Discussion by dhruvkb** [WordPress/openverse#3] [Ideas]
Requirements:
Using the project, we should be able to
- have scripts written Python, Node.js (and possibly other languages) coexist in the same repository.
- share common
  - utilities between scripts written in the same language
  - data between scripts of different languages

Proposal:
We structure the repo into language-wise directories, so that we can use individual package management for each language
```
- ccsearch-scripts/
  ├── .github/
  │    └── workflows/
  ├── python/
  │   ├── shared/ (for common Python modules)
  │   └── Pipfile and Pipfile.lock
  ├── node/
  │   ├── shared/ (for common Node.js utils)
  │   └── package.json and package-lock.json
  └── data/
       └── YAML files for shared data (such as GitHub organisation, repo names, board IDs etc.)
```

Any thought on improving the format? For example, this does not factor in how adhoc/one-off scripts would be managed, along with their dependencies.

**Comment by obulat** [comment]
This looks much better than what I had in mind (a simple pile of scripts in the repo)!

**Comment by zackkrida** [comment]
I like this structure. As for adhoc scripts, I think they can just go in a `src` or `scripts` directory within our language folders, and effectively treated the same way as scripts scheduled by workflows. 

The only difference will be that they _don't_ have associated workflows, or may require environment variables not configured in the repository, and should be noted as such.

**Comment by zackkrida** [comment]
We've followed this structure well, this repo is currently in good health.
